test-config.md

```diff
@@ -0,0 +1,56 @@
+|:---------------------|:-------------------------------------------|:------------------------------|:--------------------------------------------------|
+| `createGroup`        | -                                          | -                             | A new client for the `actor` and a new group with themselves in it are created.      |
+| `createKeyPackage`   | -                                          | `keyPackage`: packet                   | A new client for the `actor` and a new key package for them are created.      |
+| `addProposal`        | `keyPackage`: int                          | `proposal`: packet               | The `actor`’s current client sends a proposal that adds a member with the key package created during the `createKeyPackage` action with index `keyPackage`. |
+| `removeProposal`     | `removed`: string                          | `proposal`: packet               | The `actor`’s current client sends a proposal that removes the actor indicated in `removed`. |
+| `updateProposal`     | -                                          | `proposal`: packet               | The `actor`’s current client sends an update. |
+| `commit`             | `byReference`: []int                       | `commit`: packet, `welcome`: packet | The `actor`’s current client first processes the proposals created during actions identified by the indices in the list `byReference` and then they send a commit. The `byReference` list MUST NOT include proposals from `actor` (those should be committed automatically). |
+| `handleCommit`       | `commit`: int, `byReference`: []int        | `stateUpdate` (see below)                  | The `actor`’s current client first processes the proposals created during actions with indices specified in `byReference`. Then it processes the commit created during the `commit` action with index `commit` which includes them by reference. The `byReference` list MUST NOT include proposals created by `actor`. The `commit` MUST NOT have been created by `actor`.
+| `handlePendingCommit`| -                                          | `stateUpdate` (see below)                  | The `actor`’s current client accepts the commit it sent. |
+| `joinGroup`          | `welcome`: int                        | [TODO]                        | The `actor`’s current client joins the group using the welcome message created during the `commit` action index `welcome`. This creates a new session.
+| `protect`            | `applicationData`: string                  | `ciphertext`: packet             | The `actor`’s current client encrypts given data, given as b64 encoded bytes. |
+| `unprotect`          | `ciphertext`: int                          | `applicationData`: hex encoded bytes              | The `actor`’s current client decrypts the ciphertext created during the `protect` action with index `ciphertext`. The message MUST NOT have been generated by the actor. |
```

Comment:
I wonder if these would be better as `verifyProtectUnprotect`:
1. Test runner makes up a random string
2. Test runner has a member of the group protect
3. Test runner has each other member unprotect

You could either specify the sender in the script, or have the runner automatically iterate through all senders.

Reply:
I like it. If others don't oppose, I will change to this version.